src/material/select/src/select.type.ts

```diff
@@ -5,6 +5,15 @@ import { FieldType } from '@ngx-formly/material/form-field';
 @Component({
   selector: 'formly-field-mat-select',
   template: `
+    <ng-template #selectAll>
+      <mat-option (click)="toggleSelectAll()">
+        <mat-pseudo-checkbox class="mat-option-pseudo-checkbox"
+          [state]="state">
+        </mat-pseudo-checkbox>
+        {{ getSelectAllLabel() }}
```

Comment:
I think we should enforce passing the selectAll label to avoid handling the boolean check, so I propose to use:

```html
{{ to.selectAllOption }}
```